dnsapi/dns_aws.sh

```diff
@@ -152,50 +152,58 @@ dns_aws_rm() {
 
 _get_root() {
   domain=$1
-  i=2
-  p=1
-
-  if aws_rest GET "2013-04-01/hostedzone"; then
-    while true; do
-      h=$(printf "%s" "$domain" | cut -d . -f $i-100)
-      _debug2 "Checking domain: $h"
-      if [ -z "$h" ]; then
-        if _contains "$response" "<IsTruncated>true</IsTruncated>" && _contains "$response" "<NextMarker>"; then
-          _debug "IsTruncated"
-          _nextMarker="$(echo "$response" | _egrep_o "<NextMarker>.*</NextMarker>" | cut -d '>' -f 2 | cut -d '<' -f 1)"
-          _debug "NextMarker" "$_nextMarker"
-          if aws_rest GET "2013-04-01/hostedzone" "marker=$_nextMarker"; then
-            _debug "Truncated request OK"
-            i=2
-            p=1
-            continue
-          else
-            _err "Truncated request error."
-          fi
-        fi
-        #not valid
-        _err "Invalid domain"
-        return 1
-      fi
+  # Start with field 2 since each domain starts with _acme-challenge
+  # Example: _acme-challenge.www.domain.com
+  field=2
```

Comment:
don't start with field 2. 
In dns alias mode,  there is no _acme-challenge prefix at all.